src/Sarif.Multitool.FunctionalTests/TestData/UrisMustBeValid/Invalid.sarif

```diff
@@ -25,11 +25,12 @@
           ]
         }
       ],
-      "files": {
-        "fi%le:///c:/src/file.c": {
+      "files": [
```

Comment:
>files [](start = 7, length = 5)

for this file, had to spot-inject the v2 pattern, otherwise this non-transformed file doesn't deserialize. what a frankenfile! or i guess it's actually frankenstein's file

Reply:
Yes, that's similar to the issue I mentioned in my comment on the other Invalid.sarif file that you changed. The "expected" file has a couple of instances of `"message": "blah"` instead of `"message": { "text": "blah" }`. And the `PreReleaseCompatibilityTransformer` is skipping the core transformations because it's called with `forceUpdate: false`, and so those bad messages are left alone, and so the file won't deserialize as v2.

---
In reply to: [248137193](https://github.com/Microsoft/sarif-sdk/pull/1215#discussion_r248137193) [](ancestors = 248137193)